portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -99,29 +100,22 @@ private int getRequestedCommentCount(HttpServletRequest request) {
     return commentCount;
   }
 
-  public int getIdNum() {
-    int idNum = 0;
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      String idString = (String) entity.getProperty("id");
-      int id = Integer.parseInt(idString);
-      if (id > idNum) {
-        idNum = id;
-      }
-    }
-    idNum++;
-    return idNum;
-  }
+  private int idNum = -1;
```

Comment:
It would be a little bit cleaner to keep the attribute in your `ServletContext` updated and reference that every time you need to get a new ID rather than to store a copy as a variable in your servlet.

It's also possible for you to end up running multiple concurrent versions of this servlet in order to handle a larger volume of incoming requests (in theory), so if each individual servlet instance has their own `idNum`, they're going to be using duplicate IDs, which could be avoided by having all servlets read & update the same "next available ID" value in the `ServletContext`.